src/lib.rs

```diff
@@ -191,14 +239,38 @@ pub struct Machine<'tsunami> {
 impl<'t> MachineDescriptor<'t> {
     #[cfg(any(feature = "aws", feature = "azure", feature = "baremetal"))]
     #[instrument(level = "debug", skip(key_path, timeout))]
-    async fn connect_ssh(
+    async fn connect(
```

Comment:
This method is just all-around pretty weird when `openssh` is disabled, since it doesn't connect at all. I wonder if it simply shouldn't be available without the `openssh` feature? Without it, users can just use `MachineDescriptor` directly instead.

Reply:
`MachineDescriptor` is not public, only `Machine`. This method does two things: maybe connects (if `openssh` is enabled) and builds the `Machine` to be later returned to the user. Maybe rename it instead to `initialize` as suggested by @znewman01?

Reply:
I guess I'm wonder if, with this feature, `MachineDescriptor` _should_ be public, since that's basically all we expose if someone doesn't want to use `openssh`. Not entirely sure what that would look like though...

Reply:
Hmm, making `openssh` optional may be indeed a bad idea. I was trying to figure out how to get rid of `MachineDescriptor` and found out that (at least) in the AWS provider, openssh is used to check if instances are ready: https://github.com/jonhoo/tsunami/blob/master/src/providers/aws.rs#L1076-L1095